### ISSUE TITLE
ANR on sdk initialisation

### ISSUE THREAD

**Issue by marianeum** [facebook/facebook-android-sdk#574]
We're getting some ANRs caused by the facebook SDK (5.4.0) on application start when the sdk is getting initialised. We haven't managed to find steps to reproduce it, however clearing data is fixing the issues for users.

Here's a stack trace:
```
"main" prio=5 tid=1 Waiting
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x7490c790 self=0x72f7e14c00
  | sysTid=1454 nice=-10 cgrp=default sched=0/0 handle=0x737d874548
  | state=S schedstat=( 131005069 25725931 433 ) utm=7 stm=5 core=3 HZ=100
  | stack=0x7fcd167000-0x7fcd169000 stackSize=8MB
  | held mutexes=
  at java.lang.Object.wait (Object.java)
- waiting on <0x0e8b09ae> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
- locked <0x0e8b09ae> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:190)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt (AbstractQueuedSynchronizer.java:868)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly (AbstractQueuedSynchronizer.java:1021)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly (AbstractQueuedSynchronizer.java:1328)
  at java.util.concurrent.CountDownLatch.await (CountDownLatch.java:232)
  at com.facebook.internal.LockOnGetVariable.waitOnInit (LockOnGetVariable.java:64)
  at com.facebook.internal.LockOnGetVariable.getValue (LockOnGetVariable.java:54)
  at com.facebook.FacebookSdk.getCacheDir (FacebookSdk.java:954)
  at com.facebook.internal.instrument.InstrumentUtility.getInstrumentReportDir (InstrumentUtility.java:242)
  at com.facebook.internal.instrument.InstrumentUtility.listCrashReportFiles (InstrumentUtility.java:124)
  at com.facebook.internal.instrument.crashreport.CrashHandler.sendCrashReports (CrashHandler.java:101)
  at com.facebook.internal.instrument.crashreport.CrashHandler.enable (CrashHandler.java:73)
- locked <0x08ba534f> (a java.lang.Class<com.facebook.internal.instrument.crashreport.CrashHandler>)
  at com.facebook.internal.instrument.InstrumentManager.start (InstrumentManager.java:46)
  at com.facebook.FacebookSdk.sdkInitialize (FacebookSdk.java:326)
- locked <0x0ad1badc> (a java.lang.Class<com.facebook.FacebookSdk>)
  at com.facebook.marketing.internal.MarketingInitProvider.onCreate (MarketingInitProvider.java:42)
  at android.content.ContentProvider.attachInfo (ContentProvider.java:1920)
  at android.content.ContentProvider.attachInfo (ContentProvider.java:1895)
  at android.app.ActivityThread.installProvider (ActivityThread.java:6440)
  at android.app.ActivityThread.installContentProviders (ActivityThread.java:5987)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:5902)
  at android.app.ActivityThread.access$1100 (ActivityThread.java:200)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1656)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:193)
  at android.app.ActivityThread.main (ActivityThread.java:6718)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:858)
```

**Comment by cubrady** [comment]
Hi @marianeum , we found the same issue on facebook SDK 5.4.0 as well in our project, would you like to share how you solve this problem ? Thanks a lot !

**Comment by cubrady** [comment]
Answered by myself : https://github.com/facebook/facebook-android-sdk/commit/856665c5e0ede060b3a0f18af8a808aba26d4778 should fix this problem, so update the SDK to 5.5

**Comment by KylinChang** [comment]
Hi @cubrady , you can update the FBAndroidSDK to the latest version v5.5.1 to fix the issue. You can just re-export the project to update the SDK.
